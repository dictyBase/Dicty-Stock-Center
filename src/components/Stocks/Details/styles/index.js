import { makeStyles } from "@material-ui/core/styles"
import { green } from "@material-ui/core/colors"

const useStyles = makeStyles(({ palette, spacing }) => ({
  root: {
    width: "90%",
    margin: "auto",
  },
  header: {
    textAlign: "center",
  },
  backButton: {
    marginTop: "10px",
    textAlign: "left",
    color: "#5f6368",
    "&:hover": {
      backgroundColor: "transparent",
      color: "inherit",
    },
  },
  cardHeader: {
    color: "#002f5e",
  },
  details: {
    "&:nth-child(odd)": {
      backgroundColor: "#F4F6F8",
    },
    "@media (max-width: 420px)": {
      display: "block !important",
    },
  },
  listItem: {
    "@media (max-width: 420px)": {
      marginBottom: "10px",
    },
  },
  list: {
    width: "100%",
    padding: "0px !important",
  },
  availabilityCard: {
    paddingTop: "5px",
    textAlign: "center",
  },
  cartBtnRow: {
    padding: "16px",
  },
  unavailableText: {
    padding: "16px",
    textAlign: "justify",
  },
  moreStrainsCard: {
    marginTop: "16px",
    padding: "5px",
    textAlign: "center",
  },
  options: {
    paddingBottom: "8px",
    display: "flex",
    justifyContent: "center",
  },
  secondaryText: {
    paddingTop: "8px",
  },
  link: {
    textDecoration: "none",
    color: palette.primary.main,
  },
  checkoutBtn: {
    borderStyle: "none !important",
    borderRadius: "0px !important",
    fontSize: "0.9rem !important",
  },
  arrowIcon: {
    marginRight: "8px",
  },
  id: {
    display: "flex",
    justifyContent: "center",
  },
  snackbar: {
    width: "300px",
    backgroundColor: green[600],
    fontSize: "0.8rem",
    display: "flex",
    alignItems: "center",
    paddingTop: "0px",
    paddingBottom: "0px",
  },
  checkIcon: {
    fontSize: "0.8rem",
    marginRight: spacing(1),
  },
  closeIcon: {
    fontSize: "0.9rem",
  },
  unavailableIcon: {
    marginRight: spacing(2),
  },
  copyIcon: {
    fontSize: "0.8rem",
    "&:hover": {
      backgroundColor: "transparent",
      color: "rgba(0, 0, 0, 0.87)",
    },
  },
  detailsItem: {
    fontWeight: 420,
  },
  relatedItem: {
    display: "flex",
    justifyContent: "center !important",
  },
  addToCartBtn: {
    backgroundColor: green[600],
    color: "#fff",
    paddingLeft: "16px",
    paddingRight: "16px",
    "&:hover": {
      backgroundColor: green[700],
      color: "#fff",
    },
  },
  removeBtn: {
    marginTop: "5px",
  },
  quantity: {
    display: "flex",
    justifyContent: "space-evenly",
    alignItems: "center",
    padding: "16px",
  },
  maxItems: {
    color: palette.secondary.main,
  },
}))

export default useStyles
