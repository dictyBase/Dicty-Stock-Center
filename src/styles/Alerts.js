import styled from "styled-components"

export const AlertBox = styled.div`
  padding: 15px;
  margin-bottom: 20px;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #a94442;
  background-color: #f2dede;
  border-color: #ebccd1;
  text-align: center;
`

export const AlertSuccess = styled.div`
  padding: 15px;
  margin-bottom: 20px;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #3c763d;
  background-color: #dff0d8;
  border-color: #d6e9c6;
  text-align: center;
`

export const AlertInfo = styled.div`
  padding: 15px;
  margin-bottom: 20px;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #31708f;
  background-color: #d9edf7;
  border-color: #bce8f1;
`
