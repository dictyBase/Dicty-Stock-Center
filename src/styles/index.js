import styled from 'styled-components'

export const Container = styled.div`
    padding-right: 15px;
    padding-left: 15px;
    margin-right: auto;
    margin-left: auto;
    @media (min-width: 768px) {
        .container {
          width: 750px;
        }
      }
      @media (min-width: 992px) {
        .container {
          width: 970px;
        }
      }
      @media (min-width: 1200px) {
        .container {
          width: 1170px;
        }
      }
`
export const Grid = styled.div`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    min-width: 100%;
`

// export const Row = styled.div`
//     &::after {
//       content: '';
//       clear: both;
//       display: table;
//     }
// `

// function getWidthString(span) {
//     if (!span) return

//     let width = span / 12 * 100
//     return `width: ${width}%`
// }

// export const Column = styled.div`
//     float: left;
//     ${({ xs }) => (xs ? getWidthString(xs) : 'width: 100%')}

//   @media only screen and (min-width: 768px) {
//     ${({ sm }) => sm && getWidthString(sm)}
//   }

//   @media only screen and (min-width: 992px) {
//     ${({ md }) => md && getWidthString(md)}
//   }

//   @media only screen and (min-width: 1200px) {
//     ${({ lg }) => lg && getWidthString(lg)}
//   }
// `

// export const Row = styled.div`
//     display: inline-block;
//     width: 100%;

//     /* @media (min-width: ${style.breakpoint.small}) {
//       display: flex;
//     } */
// `

// export const Column = styled.div`
//     display: inline-block;
//     /* margin-bottom: ${style.margin} */
//     margin-right: 0;
//     width: 100%;

//     /* @media (min-width: ${style.breakpoint.small}) {
//       margin-right: ${(props) => props.last ? '0' : style.margin};
//       flex: ${(props) => props.colspan}
//     } */
// `
