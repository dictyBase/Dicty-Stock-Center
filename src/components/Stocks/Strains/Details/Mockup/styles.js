import { makeStyles } from "@material-ui/core/styles"

const useStyles = makeStyles({
  root: {
    width: "80%",
    margin: "auto",
  },
  header: {
    textAlign: "center",
  },
  leftCard: {},
  details: {
    "&:hover": {
      boxShadow:
        "inset 1px 0 0 #dadce0,inset -1px 0 0 #dadce0,0 1px 2px 0 rgba(60,64,67,.3),0 1px 3px 1px rgba(60,64,67,.15)",
      zIndex: 1,
    },
    "&:nth-child(odd)": {
      backgroundColor: "#eeeeee",
    },
  },
  listTitle: {},
  listContent: {},
  rightCard: {
    padding: "5px",
    textAlign: "center",
  },
  quantity: {
    display: "inline-flex",
    justifyContent: "center",
  },
  textField: {},
})

export default useStyles
