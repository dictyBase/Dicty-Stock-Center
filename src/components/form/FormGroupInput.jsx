// @flow
import React, { Component } from 'react'
import { Flex, Box } from 'rebass'
import { ControlLabel, FormGroup, FormControl, HelpBlock } from 'styles'

type Props = {
  field: Object,
  inputType: string,
  placeholder: string
}

export default class FormGroupInput extends Component<Props> {
  displayName = 'form group with input'

  static defaultProps = {
      inputType: 'text'
  }

  render() {
      const { field, inputType, placeholder } = this.props
      const hasError = field.touched && field.error

      return (
      <FormGroup>
        <Flex wrap justify="center">
          <Box w={ [1, 1 / 4, 1 / 4] } ml={ 1 } mb={ 2 } mr={ 1 }>
            <ControlLabel>{ this.props.children }</ControlLabel>
          </Box>
          <Box w={ [1, '60%', '60%'] } mr={ 1 } ml={ 1 }>
            <FormControl
              type={ inputType }
              {...field}
              placeholder={ placeholder }
            />
            { hasError && <HelpBlock>{ field.error }</HelpBlock> }
          </Box>
        </Flex>
      </FormGroup>
    )
  }
}
