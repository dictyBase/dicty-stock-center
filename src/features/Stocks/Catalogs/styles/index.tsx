import { makeStyles, Theme } from "@material-ui/core/styles"
import { blue } from "@material-ui/core/colors"

const useStyles = makeStyles((theme: Theme) => ({
  appBar: {
    backgroundColor: blue[800],
  },
  catalogPaper: {
    height: 600,
    width: "100%",
  },
  listHeaders: {
    position: "sticky",
    top: 0,
  },
  row: {
    borderBottom: "1px solid rgba(224, 224, 224, 1)",
    "&:hover": {
      backgroundColor: "#eeeeee",
      boxShadow:
        "inset 1px 0 0 #dadce0,inset -1px 0 0 #dadce0,0 1px 2px 0 rgba(60,64,67,.3),0 1px 3px 1px rgba(60,64,67,.15)",
      zIndex: 1,
    },
    "@media (max-width: 1024px)": {
      "& p": {
        fontSize: "0.75rem !important",
      },
    },
  },
  item: {
    paddingRight: theme.spacing(1),
  },
  trashBtn: {
    color: theme.palette.error.main,
  },
}))

export default useStyles
