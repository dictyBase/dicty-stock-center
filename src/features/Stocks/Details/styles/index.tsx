import { makeStyles, Theme } from "@material-ui/core/styles"

// @ts-ignore <-- necessary to include the !important attributes
const useStyles = makeStyles(({ palette, spacing }: Theme) => ({
  header: {
    textAlign: "center",
  },
  cardHeader: {
    color: "#002f5e",
    paddingRight: "8px",
  },
  details: {
    "&:nth-child(odd)": {
      backgroundColor: "#F4F6F8",
    },
    "@media (max-width: 420px)": {
      display: "block !important",
    },
  },
  listItem: {
    "@media (max-width: 420px)": {
      marginBottom: "10px",
    },
  },
  list: {
    width: "100%",
    padding: "0px !important",
  },
  id: {
    display: "flex",
    justifyContent: "center",
  },
  snackbar: {
    width: "300px",
    backgroundColor: palette.secondary.main,
    fontSize: "0.8rem",
    display: "flex",
    alignItems: "center",
    paddingTop: "0px",
    paddingBottom: "0px",
  },
  checkIcon: {
    fontSize: "0.8rem",
    marginRight: spacing(1),
  },
  closeIcon: {
    fontSize: "0.9rem",
  },
  copyIcon: {
    fontSize: "0.8rem",
    "&:hover": {
      backgroundColor: "transparent",
      color: "rgba(0, 0, 0, 0.87)",
    },
  },
  detailsItem: {
    fontWeight: 420,
  },
  maxItems: {
    color: palette.error.main,
  },
  moreInfoText: {
    marginTop: "4px",
    fontSize: "0.75rem",
    color: "#000",
  },
  error: {
    color: palette.error.main,
  },
}))

export default useStyles
