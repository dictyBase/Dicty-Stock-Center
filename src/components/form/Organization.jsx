// @flow
import React, { Component } from 'react'
import FormGroupInput from './FormGroupInput'
import { RequiredText } from 'styles'

type Props = {
    org: Object,
    group: Object
}

export default class Organization extends Component<Props> {
    displayName = 'component for organization/lab/group info';

    render() {
        const { org, group } = this.props
        return (
            <div>
                <FormGroupInput field={ org } >
                    <RequiredText title="required field">* </RequiredText>
                    Organization:
                </FormGroupInput>
                <FormGroupInput field={ group } >
                    <RequiredText title="required field">* </RequiredText>
                    Lab/Group:
                </FormGroupInput>
            </div>
        )
    }
}
