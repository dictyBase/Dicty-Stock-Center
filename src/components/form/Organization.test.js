import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import Organization from "./Organization"
import FormGroupInput from "./FormGroupInput"
import { shallow } from "enzyme"
import { RequiredText } from "styles"

describe("form/Organization", () => {
  const org = { value: "Northwestern", touched: true }
  const group = { value: "dictybase", touched: true }

  const wrapper = shallow(<Organization org={org} group={group} />)

  it("should render organization field", () => {
    expect(
      wrapper.contains(
        <FormGroupInput field={org}>
          <RequiredText title="required field">* </RequiredText>
          Organization:
        </FormGroupInput>
      )
    ).toEqual(true)
  })
  it("should render lab/group field", () => {
    expect(
      wrapper.contains(
        <FormGroupInput field={group}>
          <RequiredText title="required field">* </RequiredText>
          Lab/Group:
        </FormGroupInput>
      )
    ).toEqual(true)
  })
})

test("matching a snapshot of Organization", () => {
  const org = {
    value: "Northwestern",
    touched: true
  }
  const group = {
    value: "dictyBase",
    touched: true
  }

  const component = renderer.create(<Organization org={org} group={group} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
