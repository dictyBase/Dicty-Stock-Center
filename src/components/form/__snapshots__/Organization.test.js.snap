// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Organization 1`] = `
<div>
  <div
    className="sc-hmXxxW ggtKDs"
  >
    <div
      className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa iplmxF"
        is={null}
      >
        <label
          className="sc-ipXKqB hIQKmZ"
        >
          <span
            className="sc-fONwsr kDvBqR"
            title="required field"
          >
            * 
          </span>
          Organization:
        </label>
      </div>
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa cvzxNy"
        is={null}
      >
        <input
          className="sc-cqCuEk jbXqms"
          placeholder={undefined}
          type="text"
          value="Northwestern"
        />
      </div>
    </div>
  </div>
  <div
    className="sc-hmXxxW ggtKDs"
  >
    <div
      className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa iplmxF"
        is={null}
      >
        <label
          className="sc-ipXKqB hIQKmZ"
        >
          <span
            className="sc-fONwsr kDvBqR"
            title="required field"
          >
            * 
          </span>
          Lab/Group:
        </label>
      </div>
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa cvzxNy"
        is={null}
      >
        <input
          className="sc-cqCuEk jbXqms"
          placeholder={undefined}
          type="text"
          value="dictyBase"
        />
      </div>
    </div>
  </div>
</div>
`;
